dockstore-webservice/src/main/java/io/dockstore/webservice/resources/OrganizationResource.java

```diff
@@ -409,6 +410,31 @@ private Organization getOrganizationByIdOptionalAuth(Optional<User> user, Long o
         }
     }
 
+    @DELETE
+    @Path("/{organizationId}/deleteNonApprovedOrg")
+    @Timed
+    @UnitOfWork
+    @ApiOperation(value = "Deletes an organization that has been rejected", authorizations = { @Authorization(value = JWT_SECURITY_DEFINITION_NAME) }, response = void.class, hidden = true)
+    @Operation(operationId = "deleteRejectedOrPendingOrganization", summary = "Delete rejected organization", description = "Delete rejected organization")
+    public void deleteRejectedOrPendingOrganization(
+            @ApiParam(hidden = true) @Parameter(hidden = true, name = "user") @Auth Optional<User> user,
+            @Parameter(description = "Organization ID.", name = "organizationId", in = ParameterIn.PATH, required = true) @PathParam("organizationId") Long organizationId) {
+        Organization organization = organizationDAO.findById(organizationId);
+
+        if (organization.getStatus() == Organization.ApplicationState.APPROVED) {
```

Comment:
You are checking that it is not approved, not that it is rejected. We had some discussion about whether to also support allowing the deletion of pending organizations; if your intent is to support, which I guess it, based on the method name, then:

1. Your error message and the annotations on line 417 and 418 should reflect both deleted and rejected.
2. It would be better to explicitly check for the states we allow deletion for rather than the states we don't allow deletion for. IOW, check for the org being rejected or pending, instead of checking that is approved. This may sound silly, since they are today logically equivalent, but maybe someday we'll add a new state, I don't know, like SUSPENDED, which we wouldn't want to be deletable. This code as is would by default allow deletion of any new states.

Reply:
Reworked the messages and descriptions to better reflect on the intentions of the endpoint. Also changed the logic to only delete if the state is at PENDING or REJECTED instead of not ACCEPTED.